sdk/storage/src/core/clients/storage_account_client.rs

```diff
@@ -42,12 +54,15 @@ pub struct StorageAccountClient {
     filesystem_url: Url,
 }
 
-fn get_sas_token_parms(sas_token: &str) -> Result<Vec<(String, String)>, url::ParseError> {
+fn get_sas_token_params_with_env(
+    sas_token: &str,
+    environment: &AzureEnvironment,
```

Comment:
As mentioned in the general comments, this is inconsistent with Azure SDK guidelines that SDKs be [consistent](https://azure.github.io/azure-sdk/general_introduction.html#consistent) except when idiomatic differences would make more sense, and I don't see taking an environment with fixed URIs is idiomatic. Supplying a full URI is just as easy. Customers could also keep their own structure of common URIs and pass those as `&str` parameters.

Reply:
In general, the Azure SDK hasn't done much with environments (not in public APIs, anyway - certainly in testing) in support of Azure Stack, in which case the customer is entirely in control of their domain suffixes and endpoints in general.